src/django_components/component.py

```diff
@@ -3280,16 +3592,31 @@ def renderer(root_attributes: Optional[List[str]] = None) -> Tuple[str, Dict[str
             # Emit signal that the template is about to be rendered
             template_rendered.send(sender=template, template=template, context=context)
 
-            if template is not None:
-                # Get the component's HTML
-                html_content = template.render(context)
+            # Get the component's HTML
+            # To access the *final* output (with all its children rendered) from within `Component.on_render()`,
+            # users may convert it to a generator by including a `yield` keyword. If they do so, the part of code
+            # AFTER the yield will be called once, when the component's HTML is fully rendered.
+            #
+            # Hence we have to distinguish between the two, and pass the generator with the HTML content
+            html_content_or_generator = component.on_render(context, template)
```

Comment:
When user includes `yield` keyword in `Component.on_render()`:

```py
class Table(Component):
    def on_render(self, context, template):
        html, error = yield template.render(context)
```

Then the way Python works is that what the `on_render` method returns is actually a [Generator](https://docs.python.org/3/reference/expressions.html#yield-expressions) object.

It is this generator object which allows us to:
1. Receive the result of `template.render(context)` from the middle of the function
2. Pass back into the middle of function the result of rendering (`html, error`)
3. Internally pass the generator around.